discoveryApiResearchNowApiMapping.md

```diff
@@ -0,0 +1,436 @@
+# Discovery API - ResearchNow API mapping
+## Discovery API
+[Github](https://github.com/NYPL-discovery/discovery-api/blob/master/README.md#searching)
+### Parameters
+[GET `/v0.1/discovery/resources`](https://platformdocs.nypl.org/#/discovery/get_v0_1_discovery_resources)
+* `q` string
+* `page` string
+* `per_page` integer
+* `sort` integer
+  * "relevance", "title", "creator", "date"
+* `sort_direction` string
+  * "asc", "desc"
+  * title defaults to asc, date defaults to desc, creator defaults to asc, relevance is fixed desc
+* `search_scope` string
+  * "all", "title", "contributor", "subject", "series", "callnumber", "standard_number"
+* `filters` string
+  * "owner", "subjectLiteral", "holdingLocation", "deliveryLocation", "language", "materialType", "mediaType", "carrierType", "publisher", "contributor", "creator", "issuance", "createdYear", "dateAfte"', or "dateBefore"
+  * Specify a hash of filters to apply, where keys are from terms above
+
+## ResearchNow API
+[Github](https://github.com/NYPL/sfr-ingest-pipeline/tree/development/app/sfr-search-api#searching)
+### Parameters
+[GET `/v0.1/research-now/v3/search-api`](https://dev-platformdocs.nypl.org/#/research-now/get_v0_1_research_now_v3_search_api)
+* **`field`** (required) string
+  * "keyword", "title", "author", "standard_number" (ISBN, ISSN, LCCN and OCLC) and "subject"
+  * Defaults to "keyword"
+* **`query`** (required) string
+* `recordType`
+  * Internal record type to return with the work. Either instances or editions.
+* `page` integer
+* `per_page` integer
+* `sort`
+* `language`
+* `years`
+  * This should be formatted as {"start": year, "end": year}.
+
+<!-- <style>
+  table ul {
+    list-style: none;
+  }
+</style> -->
+
+## API Parameters Comparison
+This table lists, in the first column, the frontend features related to searching and search filters available from the frontend. The 2nd and 3rd column list out the Discovery API and ResearchNow API parameters that correlate to that frontend functionality.
+
+Single quotes (') are used for frontend terminology. Italics used to describe frontend feature. Code styling is used for parameters the respective APIs except. Doubles quotes are used for keys in Parameters that take an object. (")
```

Comment:
I'm not sure I follow all these conventions. "Doubles quotes are used for keys in Parameters that take an object.". Does `filters[materialType]` "take an object"? Or do you mean that the `filters` param itself is an object?

Relatedly, we'll need a crosswalk between values in some cases. When the DiscoveryAPI includes `filters[lang]=lang:eng`, that should trigger a RN API filter that's something like `language=English`, right? (i.e. I'm guessing that the filter value of "lang:eng" used in the DiscoveryAPI is not a valid RN filter value..)

Reply:
Thanks for the suggestions. I was not sure what convention to use. I think what you used in the comment is more helpful and easier to understand. Will update.

Also, I see ResearchNow includes an endpoint to retrieve a list of all languages. Is there a way to get a list of all language codes in Discovery API?

Reply:
On the language issue we do have ISO_3 language codes in our ElasticSearch index, so it would be very painless for us to expose those as a filter option, especially if it would simplify the work on your end to add the functionality.